File: types_of_sorting/Selection Sort.cpp
#include <iostream>
#include <vector>
using namespace std;

int main()
{
    int n;
    bool swap_check = true;
    cout << "Enter the amount of numbers to sort: ";
    cin >> n;
    int arr[n];
    cout << "Enter " << n << " numbers: ";
   // int num;

    // Input
    for (int i = 0; i < n; i++)
    {
        cin>>arr[i];
    }
    
    // Selection Sort
    
    int i, j, min_idx;  
  
    // One by one move boundary of unsorted subarray  
    for (i = 0; i < n-1; i++)  
    {  
        // Find the minimum element in unsorted array  
        min_idx = i;  
        for (j = i+1; j < n; j++)  
        if (arr[j] < arr[min_idx])  
            min_idx = j;  
  
        // Swap the found minimum element with the first element  
         int s=arr[i];
        arr[i]=arr[min_idx];
        arr[min_idx]=s;
    }
   // Output
    cout << "\nSorted Array : ";
    for (int i = 0; i < n; i++)
    {
        cout<<arr[i]<<" ";
    }
    cout<<endl;
    return 0;
} 
   

File: Programming_Languages/Python/Algorithm/Selection_Sort.py
def Selection_Sort (arr):
    for i in range(len(arr)-1):
        min,pos = arr[i],i
        for j in range(i+1,len(arr)):
            if arr[j]<min:
                min,pos = arr[j],j
        if min!=arr[i]:
            arr[i],arr[pos]=arr[pos],arr[i]
    print(arr)

# Custom Array input
# arr = list(map(int,input('Enter numbers : ').split()))
arr = [23, 89, 45, 20, 15, 90, 49, 60, 30, 95]
Selection_Sort(arr) 